src/iarc7_motion/iarc_tasks/go_to_roomba_task.py

```diff
@@ -0,0 +1,238 @@
+                """
+
+                if self._check_max_ending_roomba_range():
+                    return (TaskDone(),)
+
+                roomba_x_velocity = self._roomba_odometry.twist.twist.linear.x
+                roomba_y_velocity = self._roomba_odometry.twist.twist.linear.y
+                roomba_velocity = math.sqrt(roomba_x_velocity**2 + roomba_y_velocity**2)
+
+                x_overshoot = roomba_x_velocity/roomba_velocity * self._overshoot
+                y_overshoot = roomba_y_velocity/roomba_velocity * self._overshoot
+
+                # p-controller
+                x_vel_target = ((self._roomba_point.point.x + x_overshoot)
+                                    * self._K_X + roomba_x_velocity)
+                y_vel_target = ((self._roomba_point.point.y + y_overshoot)
+                                    * self._K_Y + roomba_y_velocity)
+
+                z_vel_target = self._z_holder.get_height_hold_response(
+                    self._drone_odometry.pose.pose.position.z)
+
+                #caps velocity
+                vel_target = math.sqrt(x_vel_target**2 + y_vel_target**2)
+
+                if vel_target > self._MAX_HORIZ_SPEED:
+                    x_vel_target = x_vel_target * (self._MAX_HORIZ_SPEED/vel_target)
+                    y_vel_target = y_vel_target * (self._MAX_HORIZ_SPEED/vel_target)
+
+                if (abs(z_vel_target) > self._MAX_Z_VELOCITY):
+                    z_vel_target =  math.copysign(self._MAX_Z_VELOCITY, z_vel_target)
+
+                desired_vel = [x_vel_target, y_vel_target, z_vel_target]
+
+                drone_vel_x = self._drone_odometry.twist.twist.linear.x
+                drone_vel_y = self._drone_odometry.twist.twist.linear.y
+                drone_vel_z = self._drone_odometry.twist.twist.linear.z
+
+                if self._current_velocity is None:
+                    self._current_velocity = [drone_vel_x, drone_vel_y, drone_vel_z]
+
+                desired_vel = self._limiter.limit_acceleration(self._current_velocity, desired_vel)
+
+                velocity = TwistStamped()
+                velocity.header.frame_id = 'level_quad'
+                velocity.header.stamp = rospy.Time.now()
+                velocity.twist.linear.x = desired_vel[0]
+                velocity.twist.linear.y = desired_vel[1]
+                velocity.twist.linear.z = desired_vel[2]
+
+                self._current_velocity = desired_vel
+
+                return (TaskRunning(), VelocityCommand(velocity))
+
+            return (TaskAborted(msg='Illegal state reached in GoTo Roomba task'),)
+
+    ## checks to see if passed in roomba id is in sight of quad
+    def _check_roomba_in_sight(self):
+        for odometry in self._roomba_array.data:
+            if odometry.child_frame_id == self._roomba_id:
+                self._roomba_odometry = odometry
+                return True
+        return False
+
+    # that the drone and roomba are both within a specified distance
```

Comment:
You don't need this function since you will be very far from the roomba when this task starts.

Reply:
The function to delete is `check_max_roomba_range()`